lib/pdex/fhir_generator.rb

```diff
@@ -102,11 +104,27 @@ def generate_organizations
 
       # Generate Practitioners and PractitionerRoles
       def generate_practitioners
-        PDEX::NPPESDataRepo.practitioners.each do |nppes_data|
-          PDEX::PractitionerGenerator.new(nppes_data).generate.each do |resource|
-            write_resource(resource)
+
+        mut = Mutex.new
+        q = SizedQueue.new(100)
+        threads = []
+        2.times do
+          threads << Thread.new do
+            while !q.closed? || (q.closed? && !q.empty?) do
+              elem = q.pop
+              next unless elem
+              PDEX::PractitionerGenerator.new(elem).generate.each do |resource|
+                write_resource(resource)
```

Comment:
I assume this is thread safe? Or is this why we have STDOUT.sync= true above?

Reply:
Yeah, it only operates on data popped from the queue and writes to the FS, so it's thread safe. the `STDOUT.sync= true` is to prevent ruby from buffering STDOUT; a log line often isn't enough to flush the buffer so you can get stale output